sql/src/main/java/io/crate/expression/scalar/string/TranslateFunction.java

```diff
@@ -0,0 +1,95 @@
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.TripleScalar;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.types.DataTypes;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+public final class TranslateFunction {
+    public static void register(ScalarFunctionModule module) {
+        var functionInfo = new FunctionInfo(new FunctionIdent("translate", ImmutableList.of(DataTypes.STRING, DataTypes.STRING, DataTypes.STRING)), DataTypes.STRING);
+        module.register(new TripleScalar<>(functionInfo, TranslateFunction::translate));
+    }
+
+    private static String translate(String text, String from, String to) {
+        if (from.isEmpty() || text.isEmpty()) {
+            return text;
+        } else {
+            return applyTranslationsToText(text, createTranslationMap(from, to));
+        }
+    }
+
+    private static HashMap<Character, Consumer<StringBuilder>> createTranslationMap(String from, String to) {
+        var translationMap = new HashMap<Character, Consumer<StringBuilder>>();
+        var fromLength = from.length();
+        var toLength = to.length();
+
+        for (int i = 0; i < fromLength; i++) {
+            Character fromChar = from.charAt(i);
+
+            checkDuplicatesInFromArg(fromChar, translationMap);
+
+            if (i < toLength) {
+                final var toChar = to.charAt(i);
+                translationMap.put(fromChar, (sb) -> sb.append(toChar));
+            } else {
+                translationMap.put(fromChar, (sb) -> {});
+            }
+        }
+        return translationMap;
+    }
+
+    private static String applyTranslationsToText(String text, HashMap<Character, Consumer<StringBuilder>> translationMap) {
+        var outputSb = new StringBuilder();
+
+        for (var ch: text.toCharArray()) {
+            if (translationMap.containsKey(ch)) {
+                translationMap.get(ch).accept(outputSb);
+            } else {
+                outputSb.append(ch);
+            }
+        }
+
+        return outputSb.toString();
+    }
+
+    /**
+     * Repeated occurrence of any char in the 'from' argument creates an ambiguity of resolving a correct translation.
+     * Therefore, such arguments treated as invalid.
+     */
+    private static void checkDuplicatesInFromArg(Character ch, HashMap<Character, Consumer<StringBuilder>> translationMap) {
+        if (translationMap.containsKey(ch)) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ENGLISH, "Argument 'from' contains duplicate characters '%s'", ch)
```

Comment:
I wonder were this behaviour is coming from, afaik PostgreSQL does not complain on duplicate `from` chars but just ignore the later ones. Anything I'm missing?

Reply:
Hi Sebastian!

A scenario with field values 
```
from -> to:
'aaa'-> '123'
```
can be handled by 

1. autoresolving:
1a. use the first occurence: a -> 1
1b. use the last occurence: a -> 3 (PostgreSQL)
2. raising exception

Please, let me know if it's important is to keep the implementation as close as pg's it can be a good reason to update the implementation to the scenario 1b. 
On the other hand, if it's not the priority, and the priority is a predictable, strict behaviour, the auto-resolving seems to look like an arbitrary implicit solution (in my opinion, sure).

Please, let me know your thoughts!

Reply:
@dvorobiov Yes we want this scalar to behave the same as in PostgreSQL, so 1a please :-)